feature/google/src/main/java/com/revenuecat/purchases/google/attribution/GoogleDeviceIdentifiersFetcher.kt

```diff
@@ -8,23 +8,31 @@ import com.google.android.gms.common.GooglePlayServicesNotAvailableException
 import com.google.android.gms.common.GooglePlayServicesRepairableException
 import com.revenuecat.purchases.common.Dispatcher
 import com.revenuecat.purchases.common.LogIntent
+import com.revenuecat.purchases.common.subscriberattributes.DeviceIdentifiersFetcher
 import com.revenuecat.purchases.common.log
+import com.revenuecat.purchases.common.subscriberattributes.SubscriberAttributeKey
 import com.revenuecat.purchases.strings.AttributionStrings
+import com.revenuecat.purchases.utils.filterNotNullValues
 import java.io.IOException
 import java.util.concurrent.TimeoutException
 
-class AttributionFetcher(
+class GoogleDeviceIdentifiersFetcher(
     private val dispatcher: Dispatcher
-) {
+) : DeviceIdentifiersFetcher {
 
-    fun getDeviceIdentifiers(
+    override fun getDeviceIdentifiers(
         applicationContext: Application,
-        completion: (advertisingID: String?, androidID: String) -> Unit
+        completion: (deviceIdentifiers: Map<String, String>) -> Unit
     ) {
-        dispatcher.enqueue(Runnable {
+        dispatcher.enqueue({
             val advertisingID: String? = getAdvertisingID(applicationContext)
             val androidID = getAndroidID(applicationContext)
-            completion(advertisingID, androidID)
+            val deviceIdentifiers = mapOf(
+                SubscriberAttributeKey.DeviceIdentifiers.GPSAdID.backendKey to advertisingID,
+                SubscriberAttributeKey.DeviceIdentifiers.AndroidID.backendKey to androidID,
+                SubscriberAttributeKey.DeviceIdentifiers.IP.backendKey to "true"
+            ).filterNotNullValues()
+            completion(deviceIdentifiers)
```

Comment:
Since Amazon doesn't have AndroidID, I figured it would be better to isolate all Google dependent attributes in the Google module.